holoviews/plotting/util.py

```diff
@@ -116,13 +132,18 @@ def compute_overlayable_zorders(obj, path=[]):
             # Skips branches of graph that collapse Overlay layers
             # to avoid adding layers that have been reduced or removed
             continue
+        elif depth is not None and depth < overlay_depth(inp):
+            # Skips branch of graph where the number of elements in an
+            # overlay has been reduced
+            continue
```

Comment:
That's not bad at and better than trying to complicate things for what I expect is a fairly rare case anyway.